internal/transport/http2_server.go

```diff
@@ -403,6 +403,10 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 			mdata[hf.Name] = append(mdata[hf.Name], hf.Value)
 			s.contentSubtype = contentSubtype
 			isGRPC = true
+
+		case "grpc-accept-encoding":
+			s.clientAdvertisedCompressors = hf.Value
```

Comment:
Doesn't this just replace the value with the latest occurring instance of the header?  Let's make sure we have a test that sets multiple compressors in the client, because I believe this needs to be a slice of strings, not a single value.

Reply:
I skipped managing multiple values as the client should ideally send `grpc-accept-encoding` once similar to `grpc-encoding`. At least the go client sends the header only once unless the user passes an additional compressor value in the outgoing context. 

I have updated the PR with changes along with test cases. It is still maintained as a string, and additional values are merged into a single comma-separated string

Reply:
Ahh, you're right, I was forgetting the list provided was expected to be a comma-separated, single header...